lib/modules/versioning/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,57 @@
+import { regEx } from '../../../util/regex';
+import { coerceString } from '../../../util/string';
+import type { GenericVersion } from '../generic';
+import { GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'aws-eks-addon';
+export const displayName = 'aws-eks-addon';
+
+export const urls = [];
+
+export const supportsRanges = false;
+
+const versionPattern = regEx(
+  '^v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?<metadata>-eksbuild\\.\\d+)?$',
+);
+
+export class AwsEKSAddonVersioningApi extends GenericVersioningApi {
```

Comment:
Consider using `RegExpVersioningApi` as base class, it would simplifify things

https://github.com/renovatebot/renovate/blob/abe61227b6c7f57cc49a79cd821a55e7f1e5556c/lib/modules/versioning/hermit/index.ts#L13-L21

Reply:
Gotcha. Looks like code simplified.